src/main/kotlin/org/jitsi/nlj/stats/DelayStats.kt

```diff
@@ -16,33 +16,115 @@
 
 package org.jitsi.nlj.stats
 
-import java.util.Arrays
 import java.util.concurrent.atomic.AtomicLong
 import java.util.concurrent.atomic.LongAdder
 import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.util.OrderedJsonObject
+import org.jitsi.utils.increaseAndGet
 
-open class DelayStats {
+open class DelayStats(
+    thresholdsNoMax: LongArray = longArrayOf(2, 5, 20, 50, 200, 500, 100)
+) {
     private val totalDelayMs = LongAdder()
     private val totalCount = LongAdder()
-    val averageDelay: Double
+    private val averageDelayMs: Double
         get() = totalDelayMs.sum() / totalCount.sum().toDouble()
-    val maxDelayMs = AtomicLong(0)
+    private val maxDelayMs = AtomicLong(0)
 
-    /* TODO: make thresholds configurable */
-    val thresholds = longArrayOf(2, 5, 20, 50, 200, 500, 1000)
-    val thresholdCounts = Array(thresholds.size + 1) { LongAdder() }
+    private val thresholds = longArrayOf(*thresholdsNoMax, Long.MAX_VALUE)
+    private val thresholdCounts = Array(thresholds.size + 1) { LongAdder() }
 
     fun addDelay(delayMs: Long) {
         if (delayMs >= 0) {
             totalDelayMs.add(delayMs)
-            if (delayMs > maxDelayMs.get()) {
-                maxDelayMs.set(delayMs)
-            }
+            maxDelayMs.increaseAndGet(delayMs)
             totalCount.increment()
 
-            val searchResult = Arrays.binarySearch(thresholds, delayMs)
-            val bucket = if (searchResult < 0) { - searchResult - 1 } else { searchResult }
-            thresholdCounts[bucket].increment()
+            findBucket(delayMs).increment()
+        }
+    }
+
+    private fun findBucket(delayMs: Long): LongAdder {
+        // The vast majority of values are in the first bucket, so linear search is likely faster than binary.
```

Comment:
Have you tested this?

Reply:
Some results here:
https://docs.google.com/spreadsheets/d/12vwnFdthVzAfUC1Z1jZAx1OWc7XGrH6CS36Yiivmp44/edit?usp=sharing

I used this to obtain them:
```
import java.util.Arrays

val thresholds = longArrayOf(2, 5, 20, 50, 200, 500, 1000, Long.MAX_VALUE)

fun findBucketBin(delayMs: Long): Int {
    return Arrays.binarySearch(thresholds, delayMs)
}
fun findBucketLin(delayMs: Long): Int {
    for (i in thresholds.indices) {
        if (delayMs <= thresholds[i]) return i
    }
    return thresholds.lastIndex
}

val iterations = 100_000_000
var key: Long = 1
println("Running $iterations iterations with key=$key, thresholds=[${thresholds.joinToString()}]")
repeat(50) {
    val start = System.currentTimeMillis()
    repeat(iterations) {
        findBucketBin(key)
    }
    println("${System.currentTimeMillis() - start}")
}

```